modules/post/linux/gather/checkvm.rb

```diff
@@ -70,12 +70,14 @@ def run
 
     # Check SCSI Driver
     if not vm
-      proc_scsi = read_file("/proc/scsi/scsi") rescue ""
-      case proc_scsi.gsub("\n", " ")
-      when /vmware/i
-        vm = "VMware"
-      when /vbox/i
-        vm = "VirtualBox"
+      if readable? ("/proc/scsi/scsi")
+        proc_scsi = read_file("/proc/scsi/scsi")
```

Comment:
Nice work @pingport80 
Thanks for fixing this
I think since read_file calls readable? again, you might want to just check for nil here instead of using readable?
e.g
```
proc_scsi = read_file("/proc/scsi/scsi")
if proc_scsci
  case proc_scsi.gsub("\n", " ")
...
```

Reply:
sure I'll do that :)